OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs

```diff
@@ -1,48 +1,64 @@
-            services.AddScoped<SearchFlightsController>();
-            services.AddScoped<ISearchFlightService, FlightService>();
-            services.AddScoped<IAirportServices, AirportServices>();
-            services.AddScoped<IFlightRepository, FlightRepository>();
-            services.AddScoped<IAirportRepository, AirportRepository>();
-            var serviceProvider = services.BuildServiceProvider();
-
-            searchFlightsController = serviceProvider.GetService<SearchFlightsController>();
+            server = new TestServer(new WebHostBuilder()
+                .UseStartup<Startup>());
+        }
+
+        [Theory]
+        [InlineData("BUD", "LTN", "2018-12-12")]
+        public async Task SearchFlightDetails_ValidRequest_ReturnsHttpStatusOK200(string startLocation, string destination, string departureDate)
+        {
+            var client = server.CreateClient();
+            var url = $"/api/SearchFlights/SearchFlightDetails?startLocation={startLocation}&endDestination={destination}&departureDate={departureDate}";
+
+            var response = await client.GetAsync(url);
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
         }
 
         [Theory]
-        [InlineData("BUD", "LTN")]
-        public void SearchFlight_IfAllSearchParametersAreValid_ReturnsStatusCodeOK200(string startLocation, string endLocation)
+        [InlineData("BUD", "LTN", "2018-12-12")]
+        public async Task SearchFlightDetails_InvalidAPIRequest_ReturnsHttpStatusNotFound404(string startLocation, string destination, string departureDate)
         {
-            var actualResult = searchFlightsController.SearchFlight(startLocation, endLocation, DateTime.Now.AddDays(1)) as OkObjectResult;
+            var client = server.CreateClient();
+            var url = $"/api/SearchFlights/Search?startLocation={startLocation}&endDestination={destination}&departureDate={departureDate}";
 
-            Assert.Equal((int)HttpStatusCode.OK, actualResult.StatusCode);
+            var response = await client.GetAsync(url);
+
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
         }
 
         [Theory]
-        [InlineData("BUD", "LTN")]
-        public void SearchFlight_IfDestinationNullOrEmpty_ThrowsValidationException(string startLocation, string endLocation)
+        [InlineData("BUD", "", "2018-12-12")]
+        public async Task SearchFlightDetails_InValidSearchParameterRequest_ThrowsValidationException(string startLocation, string destination, string departureDate)
         {
-            var result = searchFlightsController.SearchFlight(startLocation, null, DateTime.Now.AddDays(1)) as OkObjectResult;
+            var client = server.CreateClient();
+            var url = $"/api/SearchFlights/SearchFlightDetails?startLocation={startLocation}&endDestination={destination}&departureDate={departureDate}";
+
+            var response = await Assert.ThrowsAsync<ValidationException>(() => client.GetAsync(url));
```

Comment:
How is it throwing ValidationException? Only valid exceptions GetAsync throws are ArgumentNullException and HttpRequestException

https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.getasync?view=netcore-2.2

Reply:
Yep. Thanks for the link. Replaced the same